rootston/desktop.c

```diff
@@ -46,14 +46,35 @@ void view_get_size(struct roots_view *view, struct wlr_box *box) {
 	box->height = view->wlr_surface->current->height;
 }
 
+static void view_update_output(struct roots_view *view) {
+	struct roots_desktop *desktop = view->desktop;
+	struct roots_output *output = NULL, *_output;
+	struct wlr_box box;
+	view_get_size(view, &box);
+	wl_list_for_each(_output, &desktop->outputs, link) {
+		if (!wlr_output_layout_intersects(desktop->layout, _output->wlr_output,
+			view->x, view->y, view->x + box.width, view->x + box.height)) {
+			continue;
+		}
+		if (output == NULL
+				|| output->wlr_output->scale < _output->wlr_output->scale) {
+			output = _output;
+		}
+	}
+	if (output && output != view->output) {
+		view->output = output;
+		wlr_surface_send_enter(view->wlr_surface, output->wlr_output);
```

Comment:
Shouldn't we send one `enter` event for each output which intersects with the view?

Reply:
Spec: https://wayland.freedesktop.org/docs/html/apa.html#protocol-spec-wl_surface-event-enter

>Note that a surface may be overlapping with zero or more outputs.

So yeah, we should send `enter` to all outputs that intersects with the view.